tests/tracing/test_misc.py

```diff
@@ -60,6 +61,37 @@ def test_transaction_naming(sentry_init, capture_events):
     assert events[2]["transaction"] == "a"
 
 
+def test_transaction_data(sentry_init, capture_events):
+    sentry_init(traces_sample_rate=1.0)
+    events = capture_events()
+
+    with start_transaction(name="test-transaction"):
+        span_or_tx = sentry_sdk.get_current_span()
+        span_or_tx.set_data("foo", "bar")
+        with start_span(op="test-span") as span:
+            span.set_data("spanfoo", "spanbar")
+
+    transaction = events[0]
+    transaction_data = transaction["contexts"]["trace"]["data"]
+
+    assert "data" not in transaction.keys()
+    assert transaction_data == ApproxDict(
+        {
+            "foo": "bar",
+        }
+    )
```

Comment:
[nitpick]
I would find the following easier to understand than using `ApproxDict`:

```suggestion
    assert transaction_data.items() >= { "foo": "bar" }.items()
```

Here, we explicitly compare that `transaction_data` at least includes the `"foo": "bar"` key-value pair. With the `ApproxDict`, I always have to go to the `ApproxDict` class definition to remind myself what an `ApproxDict` does. 

I would prefer to move away from using `ApproxDict`, and perhaps replace existing usages with comparing the `.items()`. If you agree, I can make an issue for this.

Reply:
done.